tasks/main.yml

```diff
@@ -126,15 +101,54 @@
   when: postgresql_pg_hba_conf is defined
   notify: Restart postgresql
 
+- name: Create postgresql directory in /etc
+  file:
+    path: /etc/postgresql
+    state: directory
+    mode: '0755'
+
+- name: Prepare internal file for configuration
+  copy:
+    content: ""
+    dest: "/etc/postgresql/system-roles-internal.conf"
+    force: true
+    mode: 0600
+    owner: postgres
+    group: postgres
+  notify: Restart postgresql
+
+- name: Generate internal postgresql system-roles.conf
+  when: postgresql_server_tuning or postgresql_ssl_enable
+  block:
+    - name: Link generated conf file with server one
+      lineinfile:
+        path: /var/lib/pgsql/data/postgresql.conf
+        line: "include_if_exists = '/etc/postgresql/system-roles-internal.conf'"
+      notify: Restart postgresql
+
+    - name: Enable ssl in /etc/postgresql/system-roles-internal.conf
+      lineinfile:
+        path: /etc/postgresql/system-roles-internal.conf
+        line: "ssl = on"
```

Comment:
See above - so the file will never contain `ssl = something` or `#ssl = something`?  What happens if you run the role again?  Will it add `ssl = on` again to `/etc/postgresql/system-roles-internal.conf`?

Reply:
Every run overwrites the previous file with the actual settings.  In case there was no change in the file between runs, the content persists. So there should not be any duplicity.

Reply:
> Every run overwrites the previous file with the actual settings. In case there was no change in the file between runs, the content persists. So there should not be any duplicity.

ok - but that means the role will not be idempotent - it will report `"changed": true` every time, even if nothing needs to be changed.  You can make it idempotent by using `regexp`

Reply:
I've disabled the force override for /etc/postgresql/system-roles-internal.conf, so now it should change state only if the change was required.